common/helpers/field.js

```diff
@@ -215,60 +210,56 @@ function mapPersonToOption(person = {}) {
   }
 }
 
-function appendDependent(questions, assessmentCategory, field, key) {
-  const question = find(questions, { key })
-  let dependent = {}
+function filterAssessmentQuestion(fields, explicit = false) {
+  return ({ key }) => {
+    const field = fields[key]
 
-  if (question) {
-    if (field.explicit) {
-      dependent = {
-        field: `${question.key}__yesno`,
-        value: 'yes',
-      }
-    } else {
-      dependent = {
-        field: assessmentCategory,
-        value: question.id,
-      }
+    if (!field) {
+      return false
     }
 
-    return {
-      ...field,
-      dependent,
-    }
+    return explicit ? field.explicit : !field.explicit
```

Comment:
Wow, this is a bit confusing.
If `explicit` is true, then use `field.explicit`
if `explicit` is false, then use the opposite of `field.explicit`?